unimol/unimol/utils/conf_gen_cal_metrics.py

```diff
@@ -72,98 +74,130 @@ def GetDihedral(conf, atom_idx):
     )
 
 
-def single_conf_gen_bonds(tgt_mol, num_confs=1000, seed=42):
+def single_conf_gen(tgt_mol: Chem.Mol, num_confs: int = 1000, seed: int = 42, mmff: bool = False, randomize_angles: bool = False, threads: int = 0) -> Chem.Mol:
+    """ Generates conformers for a molecule. Functionality to support: https://arxiv.org/abs/2302.07061 """
     mol = copy.deepcopy(tgt_mol)
     mol = Chem.AddHs(mol)
     allconformers = AllChem.EmbedMultipleConfs(
-        mol, numConfs=num_confs, randomSeed=seed, clearConfs=True
+        mol, numConfs=num_confs, randomSeed=seed, clearConfs=True, numThreads=threads
     )
-    mol = Chem.RemoveHs(mol)
-    rotable_bonds = get_torsions(mol)
-    for i in range(len(allconformers)):
-        np.random.seed(i)
-        values = 3.1415926 * 2 * np.random.rand(len(rotable_bonds))
-        for idx in range(len(rotable_bonds)):
-            SetDihedral(mol.GetConformers()[i], rotable_bonds[idx], values[idx])
-        Chem.rdMolTransforms.CanonicalizeConformer(mol.GetConformers()[i])
-    return mol
-
 
-def single_conf_gen(tgt_mol, num_confs=1000, seed=42):
-    mol = copy.deepcopy(tgt_mol)
-    mol = Chem.AddHs(mol)
-    allconformers = AllChem.EmbedMultipleConfs(
-        mol, numConfs=num_confs, randomSeed=seed, clearConfs=True
-    )
-    sz = len(allconformers)
-    for i in range(sz):
-        try:
-            AllChem.MMFFOptimizeMolecule(mol, confId=i)
-        except:
-            continue
+    # WARNING! this might change the molecule stereochemistry
```

Comment:
explicit warning. code still here for backwards compatibility

Reply:
The main reason why this happens is bc rdkit's function changes the `i,j,k,l` torsion and **rotates all points linked to `l`** whereas my function in 6+T PR rotates all points linked to `k`. This prevents a change in stereochemistry when 2 different atoms are linked to `k`, which the RDKit function doesn't catch. 

If we want to preserve this functionality, we might want to apply the *same torsion delta to all `i,j,k,l` sharing the same `j,k`* to ensure stereochemistry is not changed